File: app/core/types/predicate.py
from enum import Enum


class Predicate(Enum):
    """Predicate operations for field comparisons."""
    EQUALS = "="
    NOT_EQUALS = "!="
    GREATER_THAN = ">"
    LESS_THAN = "<"
    GREATER_THAN_OR_EQ = ">="
    LESS_THAN_OR_EQ = "<="
    LIKE = "LIKE"

File: pkg/types/predicate.go
package types

type Predicate int

const (
	Equals Predicate = iota
	LessThan
	GreaterThan
	LessThanOrEqual
	GreaterThanOrEqual
	NotEqual
	Like
)

func (p Predicate) String() string {
	switch p {
	case Equals:
		return "="

	case LessThan:
		return "<"

	case GreaterThan:
		return ">"

	case LessThanOrEqual:
		return "<="

	case GreaterThanOrEqual:
		return ">="

	case NotEqual:
		return "!="

	case Like:
		return "LIKE"

	default:
		return "UNKNOWN"
	}
}
